### DISCUSSION TITLE
May multiple community collab audio call

### DISCUSSION THREAD

**Discussion by eddiejaoude** [EddieHubCommunity/support#1757] [Community Call]
We can talk about

- what tech you are working with 
- what you have learnt recently
- any questions
- ... anything else too 🤓 

Leave your comments below 👇

**Comment by kaiwalyakoparkar** [comment]
[ Q ] When can we call ourselves Software Engineer?

**Reply by himanshu007-creator** [reply]
If you can do "Hello world" in any language without any bugs😀
credits @eddiejaoude

**Reply by eddiejaoude** [reply]
I believe if you are all here, you are software engineers 🔥

**Reply by rohan-kulkarni-25** [reply]
I think If you can get what issue is there in your code !! You are a Software Engineer !!

**Reply by iason9** [reply]
the path to becoming a software engineer is to believe that you are one and acting upon it to actually be it in the end of road, also it is relative as you can never acquire all the knowledge there is out there!

**Reply by unnati914** [reply]
I think when you are perfect in dsa,community,and you are contributing to open source that is when you are called a software engineer

**Comment by naomi-lgbt** [comment]
Drop your introduction here!

**Reply by verma-kunal** [reply]
Hey everyone! 👋
I'm Kunal Verma, a freshman pursuing Bachelor's in IT from India. I am a full-stack web development enthusiast and also passionate about Open-source! Really like to interact with folks and be a part of awesome communities and learn more n more!

**Reply by rohan-kulkarni-25** [reply]
Hey Folks !! I am Rohan ! Fresher from SITS PUNE !! Learning Web-development ( Front -end ) !! The main focus to have fun with JS & React currently !! Loves to collaborate and contribute to the community !! Learning Java as second Langauge ( 1st Cpp ) !! For now Development at Focus !

**Reply by Akshu-on-github** [reply]
Hello everyone!
I'm Akshu, a student of CSE from India. I'm passionate about IoT and OpenSource! I've been infected by the EddieHub bug and strongly believe in collaboration over code 😄 
My hands are full with college at the moment, but I'm planning on writing a few tech blogs and articles

**Reply by metaverde** [reply]
Hey!
I'm so happy to have made a community call finally.
I'm active in several DAOs.
Still finding my place, still intermittently working at being a Solidity developer.

**Reply by Vyvy-vi** [reply]
Hey y'all
Vyvy-vi here :)
Your friendly neighbourhood 🥔 

I like messing with bots, mainly work with Python, but learning new stuff every now and then. Github and inclusive programming environments are things I love :)

**Reply by himanshu007-creator** [reply]
Hi!
i am Himanshu, a sophomore from Punjab (INDIA). I pursue majors in electrics, but passion in computer science😀. I can be mostly found on GitHub , like always, Its my best place to be. I like to help people get started with opensource ( from scratch ) and have helped 200+ students send their first PR.
I believe opensource is the future🎉

**Reply by vijay-jaisankar** [reply]
Hi, I'm Vijay.

I'm currently a computer science sophomore at IIIT Bangalore, and a developer-cum-core member at Zense, our uni's software engineering club.
Additionally, I'm an organiser at TEDxIIITBangalore, an editor of the 8Bit@IIITB Magazine, and a member of the university's Debate Club,DebSoc@IIITB. 

I love reading, writing, speaking, and understanding more about the world around us. 
Conversations over Iced tea, tennis, and hackathons are favorite activities to indulge in :)

**Reply by VarunS20** [reply]
Hello everyone, I am Varun , I am an undergrad majoring in computer science, and I am absolute beginner to open source
Really gald to be a part of the the discussion hoping to learn from you all

**Reply by kaiwalyakoparkar** [reply]
Hello Everyone :wave:,
I am Kaiwalya Koparkar, I am a second-year computer engineering student in Nashik, India. I love web technologies and full-stack development. I am exploring DevOps and AI/ML recently. I love to solve algorithmic questions and love to play around with different techs. I am glad that I joined EddieHub

**Reply by ghost** [reply]
Hey Folks, I am Krish a passionate web developer from India. Looking ahead to learn full stack. Started out programming to compete with elder friend's discord bots 😅 I love to make new friends and learn new things 😊

**Comment by naomi-lgbt** [comment]
In this one, tell us something good! Share an achievement or a win!

**Reply by himanshu007-creator** [reply]
I participated in hack.push.commit Paris today and got 3 PR's merged🎉
and won some freebies as well🙂

**Reply by naomi-lgbt** [reply]
That is awesome! Congrats!

**Reply by himanshu007-creator** [reply]
thnx :)

**Reply by Vyvy-vi** [reply]
That's great! :)

**Reply by rohan-kulkarni-25** [reply]
Today Completed the first 1000 commits  !!  Added 40K + Line code in @himanshu007-creator  Project of creating a website for NGO !  Crossed 200 Twitter friends !!

**Reply by himanshu007-creator** [reply]
Awesome @rohank2502

**Reply by ghost** [reply]
An Project Of Mine: [Reseter.css](https://github.com/krishdevdb/reseter.css) Reached 1m+ JsDelivr Hits 🥳

**Comment by naomi-lgbt** [comment]
And for this one, what would you like to see our community do to improve?

**Reply by slayer321** [reply]
Hey, I am sachin, and pre-final year engineering student. Looking to collaborate and learn from this community.

**Reply by Akshu-on-github** [reply]
I'd like to see more convo on wellbeing... like how to avoid bad wrists, headaches and stress in general. I find it difficult to juggle between work and play so to speak

**Reply by ghost** [reply]
Want to see more of wellbeing and fun activities

**Comment by naomi-lgbt** [comment]
Do you have any cool projects you want to show off? Drop a link in this thread!

**Reply by slayer321** [reply]
not yet still working on it

**Reply by RutvikJ77** [reply]
How about an inspirational bot guiding you along your journey on Twitter. Here is the repo for it https://github.com/RutvikJ77/Inspiquote 💯🔥

**Reply by ghost** [reply]
https://github.com/krishdevdb/reseter.css

**Comment by ashwinexe** [comment]
[Q]  How can I be a better speaker?

**Reply by RutvikJ77** [reply]
In my opinion, it's all about practice 😁

**Reply by naomi-lgbt** [reply]
Practise practise practise.

Join our community calls, hop on Radio Eddie, join @kaiwalyakoparkar's public speaking calls.

**Reply by slayer321** [reply]
Thank you for asking this question, I was also going to ask this question.

**Reply by ashwinexe** [reply]
Thanks for the reply. I'll make sure I practice consistently 😁

**Reply by ghost** [reply]
It is just the game of mind, be confident

**Comment by kaiwalyakoparkar** [comment]
[ Q ] If you had to start as a Developer again fresh then how would you do that?

**Reply by naomi-lgbt** [reply]
Join EddieHub from day one. 🙃

**Reply by rohan-kulkarni-25** [reply]
> 
> 
> Join EddieHub from day one. 🙃

Same !!

**Reply by ashwinexe** [reply]
Just like @nhcarrigan said in community call "I would have joined eddieHub and other amazing communities from day 0"

**Reply by ghost** [reply]
> Join EddieHub from day one. 🙃
Same here as well 🎉 

Plus I would have gone to freecodecamp from day one

**Comment by kaiwalyakoparkar** [comment]
[ Q ] If you get a chance to go back in time and undo your mistakes what mistakes would it be?

**Reply by naomi-lgbt** [reply]
None. Every mistake I have made has lead me to this point in my life, and I am right where I need to be.

**Reply by himanshu007-creator** [reply]
Choose computers over physical education in my secondary classes 😭😭

**Reply by iason9** [reply]
yes @nhcarrigan  exactly the present moment you are in it now is the best good outcome of probabilities that you could ever have!

**Reply by eddiejaoude** [reply]
I would start everything sooner!!

- start open source sooner
- start my YouTube channel sooner
- start EddieHub sooner, so I can geek out with you all even sooner

**Reply by iason9** [reply]
> I would start everything sooner!!
> 
>     * start open source sooner
> 
>     * start my YouTube channel sooner
> 
>     * start EddieHub sooner, so I can geek out with you all even sooner

when the opportunity presents itself to you and you are not ready for it you will think it's the badest opportunity ever and you will even reject it! so I think you started at the right time the exact moment you were ready to code and do open source and everything else. it's sometimes hard for me to believe it too but I think we all should believe it

**Reply by slayer321** [reply]
> Choose computers over physical education in my secondary classes 😭😭

same for me I took electrical as my optional subject .... I wish I must have taken CS as my options subject

**Reply by ghost** [reply]
Go over computers instead of science + Solve my mistake of not learning coding from starting of my tech story :sob:

**Comment by kaiwalyakoparkar** [comment]
[ Q ] What are your biggest regrets as a developer? (What is the thing that you think would have been beneficial if you have known it earlier)?

**Reply by naomi-lgbt** [reply]
Also none. I cannot change the past, so I don't spend the emotional energy on regret. I learn from my past obstacles and use that learning to improve.

**Reply by Vyvy-vi** [reply]
Not many (I don't keep regrets), but I wish that maybe I could have done some more low level stuff way back. (That stuff really attracts me, while I'm doing high level stuff :P )

**Reply by naomi-lgbt** [reply]
That's shiny object syndrome :P

**Reply by rohan-kulkarni-25** [reply]
I think mistakes are the first step of success ! we cant make anything perfect on the first try so !! Try Try Try

**Reply by Vyvy-vi** [reply]
> That's shiny object syndrome :P

More than shiny-ness, it's more like - I wish I can understand more of that stuff, and maybe accelerate my basics before picking up high-level stuff. (I kinda used to do some hardware stuff using Arduino, and that's something I haven't touched in a while. But Idk I would just love to understand what is happening under the hood in my code, how the computer manages it... Something that I could understand if I maybe gave more time to C or maybe learnt Assembly)

But, I guess I still have time - Might get there in future 🙃

**Reply by ghost** [reply]
My mistake of not learning coding from starting of my tech story :sob:

**Comment by abhishree07** [comment]
[ Q ] How do you get started with technical skills so that you can contribute back to the community in a wholesome way ?

**Reply by naomi-lgbt** [reply]
Not 100% sure what you mean by technical skills, but...

You don't need to be an expert developer to contribute to our community. Everyone has value to bring, even if this is your *first* day as a developer.

**Reply by eddiejaoude** [reply]
- I think someone who is 1 week into their journey, there is someone who is only 1 day (also a great way to learn and improve)
- everyone is my mentor, I learn from all of you every day ❤️

**Reply by ghost** [reply]
> You don't need to be an expert developer to contribute to our community. Everyone has value to bring, even if this is your first day as a developer.
+1

btw I would have learnt how to contribute to github, so that I could add value from day 1

**Comment by eddiejaoude** [comment]
What more can the EddieHub community do to help you?

**Reply by rohan-kulkarni-25** [reply]
We can start our Open Source  Program !! 🤔  This will be a great experience for all Eddie Team to come up with a platform where new developers will come and start their Open source journey !!

**Reply by naomi-lgbt** [reply]
Would folks find it beneficial to have a `helper` role in the Discord? The idea being this is a role you can ping when you have a question and no one is around to answer it.

**Reply by ashwinexe** [reply]
Is it possible to host a opensource themed hackathon by EddieHub?

**Reply by naomi-lgbt** [reply]
When is @SaraJaoude joining our community calls? 😁

**Reply by abhishree07** [reply]
Maybe discuss project ideas or hands on stuff. I am being very random

**Reply by kaiwalyakoparkar** [reply]
While watching your tutorial videos I feel a knowledge gap between you and me! I would love it if the tutorials resolve that gap!

**Reply by verma-kunal** [reply]
We can conduct like `Open source workshops` for all the community members, where we can have various hands-on activities or maybe various interesting tasks to complete!
These can be done monthly or as decided by the community members!!😃

**Reply by Vyvy-vi** [reply]
I think that role might be great. We could have the role as a notif for people that would like to get pinged, and that way um people can maybe get help faster

(It can also be a way to somewhat recognise people that help out, and people can opt-in to get it)

**Reply by SaraJaoude** [reply]
> When is @SaraJaoude joining our community calls? 😁

Very soon @nhcarrigan I promise! The move to Portugal is taking up most of my time...especially making sure all of Eddie's stuff can get over there :) As soon as we are settled I will be joining!

**Reply by ghost** [reply]
In collaboration + chatting (slangs are my biggest enemies)

**Comment by Vyvy-vi** [comment]
[ Q ] From a semantics, understanding and interest point of view, is there anything wrong, with planning on learning multiple languages and making stuff using them over a scale of 5-6 years? OR would you recommend sticking to 1-3 and just making stuff with those 1-3?

**Reply by naomi-lgbt** [reply]
Not at all. Learning new things is good.

I think it is important to take the time to properly understand the language if you enjoy it and want to use it. But it's totally fine to explore multiple options - I just don't recommend doing them all at once.

**Reply by eddiejaoude** [reply]
I think it is good to find a balance. Everyone is different, some can context switch better. But being a specialist in a smaller number is better.

I try to have a T shape learning, be aware of many things, but go deep into a few

**Reply by iason9** [reply]
the funny thing of choosing what you enjoy and working in it is that eventually, you will get that job in that field that you enjoy so much!

**Reply by ghost** [reply]
I would go more deeper into my favorite ones, but I will give all a go so that I'm not the last runner up, when my team works with it

**Comment by verma-kunal** [comment]
[ Q ] I am trying to get involved more and more in the community, like participating in the conversations & welcoming new folks. Honestly, I really enjoy doing that, but sometimes I feel that i kind of get too involved a lot and not able to complete my daily goals & developing my skills! This sometimes makes me feel demotivated & i think thats not a good way, as developing my skills is also as important for me as well as if I wanna guide someone as well !!
So, how can I manage both & contribute equally to both ?!

**Reply by naomi-lgbt** [reply]
What helps me is to remember that you are not *required* to give back to the community 24/7. Everyone has different things going on in their lives - your contributions are valid, even if you feel like you aren't making many.

Remember to take care of what you need to take care of to succeed.

**Reply by eddiejaoude** [reply]
Finding a balance is important. I find this balance does shift depending on what is going on and how I feel.

Important to:
- learn
- collaborate
- rest

**Reply by ghost** [reply]
> Finding a balance is important. I find this balance does shift depending on what is going on and how I feel.

> Important to:

> learn
collaborate
rest

+ infinity

**Comment by VarunS20** [comment]
How would I get started with open source as Eddie said now  sooner

**Reply by rohan-kulkarni-25** [reply]
You can Check good first issue from section and work on those !! And get 🟩 !!

**Reply by naomi-lgbt** [reply]
For starting your own open-source project:

Put it on GitHub! Don't wait until your code is "ready" (hint: it never is). Put your code up on the repo as soon as you write it - you'd be surprised how helpful the community is.

For contributing to other open-source projects:

Read the documentation, read the open issues, and then find one that interests you and say "Hey can I work on this?"

**Reply by unnati914** [reply]
True every small contribution counts

**Reply by eddiejaoude** [reply]
Ask questions is a great way to get started, questions can also add value.

- try to run the project from the documentation might have missing steps that you can improve
- many projects are missing automated tests
- projects have cli commands, these could be run on a GitHub Action

**Reply by ghost** [reply]
+100 to everyone

**Comment by RutvikJ77** [comment]
Q] What mistakes should one be aware of when starting with Freelancing?

**Reply by naomi-lgbt** [reply]
It's all about the pricing.

You need to be real with how valuable your time is - many new freelancers undervalue their time and effort. Also, remember that you should not work 24/7 - if you think a project will take 80 hours of work, don't promise to deliver it within 80 hours. Give yourself the time you need to handle other things too.

**Reply by RutvikJ77** [reply]
That's soo true. Thanks for your input 😁

**Reply by kaiwalyakoparkar** [reply]
Also, get your legal things done with the client (Like bonds and agreement) and if possible take half payment beforehand for every new client. (Many clients cheated the teams after getting the source code)

**Reply by eddiejaoude** [reply]
- use templates for documents and put your name/logo on it to look professional
- always follow up every conversation with a list of actions and notes
- new clients pay 50% up front

**Comment by abhishree07** [comment]
How do you get started with Linux. I am like intimidated by it and cmd stuff.

**Reply by naomi-lgbt** [reply]
I started with Linux by installing Windows Subsystems for Linux (WSL2). This allows me to explore and work within the UNIX environment, while still having my full Windows experience. That way, if I get stuck with something, I'm not fighting with my inexperience when trying to look things up.

**Reply by Vyvy-vi** [reply]
I'll also highly suggest taking out the Linux or UNIX-based shell for a spin, before installing/dual-booting linux, to get used to the command line(esp. with *nix systems). For a start, something as simple as - making files, making folders, moving the file around folder to folder, get the contents of the file, maybe use git to commit that to github; could give you some comfort and you can build upon this.

**Comment by unnati914** [comment]
How to learn from mistakes?

**Reply by naomi-lgbt** [reply]
Step one is to **own it**. If you don't take ownership of and responsibility for your mistakes, you cannot do the introspection needed to learn from and move forward.

Then, talk about it - whether it's in our community, with a friend, or with your cat. Talk about the mistake - what lead to the mistake, why did you make the mistake, how can you prevent the mistake in the future.

**Comment by SUCHITRAGIRI** [comment]
How to maintain your interest in a particular thing, because in my case I lose my interest very quickly?

**Reply by naomi-lgbt** [reply]
IMHO, you've got to make it interesting on your own.

I find my interest remains when I am actively building and playing with things. If I learn a new skill, like Angular, I go build stuff with it. I choose projects that tie in to my other interests, so that my passion can fuel my development.

**Reply by slayer321** [reply]
I think that there are times we lose interest in a particular topic. what  I do is just take a break from that and then start it again

**Reply by ashwinexe** [reply]
I have also lost interest in work I love to do (this includes coding too). but I make sure I'm consistent (at least everyday). In my opinion its natural for interest to go up and down with time, but if we're just aware of the fact that its the interest which is the root of my unproductivity,  we can solve it much faster.

**Reply by SUCHITRAGIRI** [reply]
@nhcarrigan Thanks for response:)
But sometime what happens to me is if something goes wrong with whatever I do as a 1st interest and that moment of time if other thing excites me I left the previous one!
How to avoid this kind of destruction or how to bear with 1st interset until it complete?

**Reply by naomi-lgbt** [reply]
Don't force it. If you force it, you'll burn out.

It's totally okay to bounce between projects. I do it all the time. Stuck on a bug? Go work on something else for a while, then come back with fresh eyes.

**Reply by iason9** [reply]
Great replies @nhcarrigan Nice work!

**Comment by Vyvy-vi** [comment]
[ Q ] What would be the methodology to go about with projects? Is making a ton of projects while learning optimal, or is it suggested to make some specific projects, "for the sake of the projects, rather than learning + making projects + practice/brushing up on something"? Is it bad that you don't have a "magnum-opus", or projects that aren't really "used" by other people?

**Reply by naomi-lgbt** [reply]
Becca is my big project. I'm doomed if Discord ever shuts down. 🤣 

I don't think you need to have projects that are the Next Big Thing:tm:, you need to have projects that demonstrate your abilities and interests. You're being hired for your *development* skills, not your marketing skills. 😉

**Comment by WebShivam** [comment]
[Q] How can I be good at collaboration on GitHub issues and discussions? I want to communicate and share my thoughts on Open Source Projects and other places like Discord and I want to sound friendly and inclusive but I feel I really struggle with it. How can I improve/learn it 🤔  
>During most cases I feel like I'm not sounding friendly, I feel my communication skills are really bad but I don't know how can I improve

**Reply by naomi-lgbt** [reply]
I mean, I'd say your communication skills are great - you did just write an awesome question here 😁 

The important thing is to keep an open mind. I'll dig up some resources on inclusive language, too, at some point.

**Reply by WebShivam** [reply]
Yeah Thank you so much resources would be great help 😄

>I spent whole 5 minutes to make this question look good don't know how much good it is now 😁

**Comment by verma-kunal** [comment]
[ Q ] Just a short question, whats the difference between **Developer Advocate** & **DevRel** Roles in tech. ? These two had somewhat the same explanation when i searched, and I got confused !!

**Reply by kaiwalyakoparkar** [reply]
+ @verma-kunal What is Avengalist

**Reply by verma-kunal** [reply]
Yeah! this is another term that comes up, although I am not sure about that as well !! 😕

**Reply by naomi-lgbt** [reply]
I think a lot of it is semantics. Much like our "developer/engineer" discussion earlier.

**Comment by vijay-jaisankar** [comment]
[Q] *Tech writing block* is something I struggle with _all_ the time. Striking a balance between sounding objective and technical and sounding friendly and engaging seems to be very hard. At times, the end product seems pretentious at best; almost an attempt to be too friendly. 

Do you have any tips to help solve this?

Thanks!

**Reply by naomi-lgbt** [reply]
This helped me a lot: https://www.freecodecamp.org/news/developer-news-style-guide/

**Reply by vijay-jaisankar** [reply]
Thanks a lot @nhcarrigan :)

**Reply by metaverde** [reply]
I don't have any suggestions, but it's fantastic that you're aware and conscientious.
Maybe try reading it out loud and maybe even recording it and see if that helps rewording things or setting tone?
